packages/style-engine/class-wp-style-engine.php

```diff
@@ -535,6 +550,41 @@ protected static function get_individual_property_css_declarations( $style_value
 		return $css_declarations;
 	}
 
+	/**
+	 * Style value parser that constructs a CSS definition array comprising a single CSS property and value.
+	 * If the provided value is an array containing a `url` property, the function will return a CSS definition array
+	 * with a single property and value, with `url` escaped and injected into a CSS `url()` function,
+	 * e.g., array( 'background-image' => "url( '...' )" ).
+	 *
+	 * @param array $style_value      A single raw style value from $block_styles array.
+	 * @param array $style_definition A single style definition from BLOCK_STYLE_DEFINITIONS_METADATA.
+	 *
+	 * @return string[] An associative array of CSS definitions, e.g., array( "$property" => "$value", "$property" => "$value" ).
+	 */
+	protected static function get_url_or_value_css_declaration( $style_value, $style_definition ) {
+		if ( empty( $style_value ) ) {
+			return array();
+		}
+
+		$css_declarations = array();
+
+		if ( isset( $style_definition['property_keys']['default'] ) ) {
+			$value = null;
+
+			if ( ! empty( $style_value['url'] ) ) {
+				$value = "url('" . $style_value['url'] . "')";
+			} elseif ( is_string( $style_value ) ) {
+				$value = $style_value;
```

Comment:
I like! 

So this could handle both of these scenarios?

```php
$media_block_styles['backgroundImage']['url'] = 'http://some/image.gif';

// or

$media_block_styles['backgroundImage'] = 'url(http://some/image.gif)';


gutenberg_style_engine_get_styles( array( 'media' => $media_block_styles ) 
```

I'm thinking of cases where it might be useful to extend the `background-image` value, e.g., 

```css
background-image: url(http://some/image.gif),
                  url(http://some/image2.gif);
```

Reply:
Yep, that's what I had in mind! We'd also support `linear-gradient` and other values for this property that way, too.

In future follow-ups it'd be fun to see how we could have layers of each of the different background properties, so you might have an array of background styles and then the style engine could figure out how to output them all in the right sequence (e.g. `background-image` as in your example, with corresponding `background-position` values, for example).